brooklyn-tosca-transformer/src/main/java/io/cloudsoft/tosca/a4c/brooklyn/Alien4CloudFacade.java

```diff
@@ -381,21 +394,37 @@ public boolean isDerivedFrom(String nodeId, Alien4CloudApplication toscaApplicat
         return getTemplatePropertyObjects(relationshipTemplate, paasNodeTemplate, builtPaaSNodeTemplates, toscaApplication.getKeywordMap(nodeTemplate, relationshipTemplate));
     }
 
-    private String resolveAttribute(Map.Entry<String, IValue> attribute, Alien4CloudApplication toscaApplication, PaaSNodeTemplate paaSNodeTemplate, Map<String, PaaSNodeTemplate> allNodes) {
-        String returnValue = FunctionEvaluator.parseAttribute(
-                attribute.getKey(),
-                attribute.getValue(),
-                toscaApplication.getTopology(),
-                ImmutableMap.<String, Map<String, InstanceInformation>>of(),
-                "",
-                paaSNodeTemplate,
-                allNodes);
-        // Alien4Cloud returns the attribute key surrounded by <> if it does not evaluate to anything
-        // We turn this to the empty String to avoid populating sensors.
-        if (returnValue.equals("<" + attribute.getKey() + ">")) {
-            return "";
+    private Object resolveAttribute(Map.Entry<String, IValue> attribute, Alien4CloudApplication toscaApplication, String nodeId, PaaSNodeTemplate paaSNodeTemplate, Map<String, PaaSNodeTemplate> allNodes) {
+        Map<String, PaaSNodeTemplate> builtPaaSNodeTemplates = getAllNodes(toscaApplication);
+        PaaSNodeTemplate paasNodeTemplate = builtPaaSNodeTemplates.get(nodeId);
+        IValue attributeValue = attribute.getValue();
+        if (attributeValue instanceof AttributeDefinition) {
+            String defaultValue = ((AttributeDefinition) attribute.getValue()).getDefault();
+            if (Strings.isBlank(defaultValue)) {
+                return null;
```

Comment:
The trouble we have with allowing empty strings, is that Alien4Cloud puts empty strings for the Brooklyn entities, and these in turn are put into the sensors but the type is different e.g JMX port on Tomcat.  I think it is easier to prevent these values from being used in the creation of Static Sensors.